### DISCUSSION TITLE
Is it possible to add more supported language

### DISCUSSION THREAD

**Discussion by cyanxiao** [retextjs/retext#75] [Q&A] [answered]
I'm recently working on a non-Latin language to CLNST library. Once it's finished, I decide to submit a PR to add more language support to retext. However, I'm not sure whether retext is currently constrained to Dutch / English / Latin and no more new languages are allowed.

**Comment by wooorm** [comment, marked as answer]
Yeah, it is possible. It is somewhat difficult though.
See `parse-dutch` and `parse-english`, which add a couple of rules on top of `parse-latin`. You could create something based on `parse-latin` as well, if you’re say, dealing with French. For languages using other scripts, it’s going to get tougher.
For retext, which includes many wordlist based rules, you’d need to translate those wordlists!

**Reply by cyanxiao** [reply]
Thank you! Actually I'm implementing Chinese & Japanese to CLNST so it may not base on `parse-latin` and use other techniques instead. I hope I can overcome the difficulties 🥲
